src/web/routes/application/middleware/handle-post.js

```diff
@@ -26,7 +26,8 @@ const handlePost = (steps, step) => (req, res, next) => {
       stateMachine.dispatch(INVALIDATE_REVIEW, req)
     }
 
-    req.session.nextAllowedStep = stateMachine.dispatch(GET_NEXT_PATH, req, steps)
+    const nextAllowedPath = stateMachine.dispatch(GET_NEXT_PATH, req, steps)
+    stateMachine.dispatch(SET_NEXT_ALLOWED_PATH, req, nextAllowedPath)
```

Comment:
These two lines are duplicated here and in `terms-and-conditions/post.js` should they be moved as a function into the state machine?

Reply:
yes I thought about that .. in the end I decided it was too much of an abstraction; I don't think the state machine should care about an implementation detail such as "set `nextAllowedStep` from the result of calling `getNextPath` for the active state".

This would involve complicating the state machine for the sake of a small amount of repeated code in middleware. I also think that the repeated code in middleware is a lot clearer than abstracting this to the state machine.

Ultimately I think this abstraction would make the code harder to understand - feel free to disagree!!

Reply:
```
// State machine
...
setNextAllowedPathFromNextPath: (req, steps) => {
  const nextAllowedPath = stateMachine.dispatch(GET_NEXT_PATH, req, steps, req.path)
  stateMachine.dispatch(SET_NEXT_ALLOWED_PATH, req, nextAllowedPath)
}
...

// Calling code
stateMachine.dispatch(SET_NEXT_ALLOWED_PATH_FROM_NEXT_PATH, req, steps)
```
.. maybe this is OK, but it feels messy and not sure we gain much except extra unit tests.

Reply:
As long as you've considered it, then I'm happy to keep it as it is if it looks like an abstraction too far.

Reply:
Having thought about this again, there's probably a distinction to be made between setting the next allowed path directly and just incrementing (can't think of a better word) the next allowed path to the next path in the sequence.

Reply:
@mikeSmith08 I've implemented a refactor - can you look at the last commit and see what you think?

Reply:
Yep, looks good to me, that's quite tidy.